Change code examples in tf.strings.format to use TF 2.x

yongtang opened the pull request:
The code examples in tf.strings.format still uses 1.x (`tf.compat.v1`)
which is not appropriate now.

This PR changes code examples in tf.strings.format to use TF 2.x.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Change code examples in tf.strings.format to use TF 2.x

The code examples in tf.strings.format still uses 1.x (`tf.compat.v1`)
which is not appropriate now.

This PR changes code examples in tf.strings.format to use TF 2.x.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

mihaimaruseac reviewed (changes requested):
LGTM but let's convert to doctests using `>>> ` format: https://www.tensorflow.org/community/contribute/docs_ref

yongtang committed:
Change to use >>> and ... in docstring format, based on review comment

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

mihaimaruseac commented on tensorflow/python/ops/string_ops.py:
```diff
@@ -123,29 +123,26 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
-
-    assert(formatted == expected)
+    >>> tensor = tf.range(10)
+    >>> formatted = tf.strings.format("tensor: {}, suffix", tensor)
+    >>> expected = "tensor: [0 1 2 ... 7 8 9], suffix"
+    >>> assert(formatted == expected)
```
We don't need the assertion written explicitly. The following should work (modulo some output change):

```suggestion
    >>> formatted = tf.strings.format("tensor: {}, suffix", tensor)
    "tensor: [0 1 2 ... 7 8 9], suffix"
```

We should do the same for the other doctests below

yongtang committed:
Remove explicit assert

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

mihaimaruseac commented on tensorflow/python/ops/string_ops.py:
```diff
@@ -123,35 +123,20 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
-
-        assert(out.decode() == expected)
+    >>> tensor = tf.range(10)
+    >>> formatted = tf.strings.format("tensor: {}, suffix", tensor)
+    >>> print(formatted)
```
```suggestion
    >>> tf.strings.format("tensor: {}, suffix", tensor)
```

mihaimaruseac commented on tensorflow/python/ops/string_ops.py:
```diff
@@ -123,35 +123,20 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
-        assert(out.decode() == expected)
+    >>> tensor_one = tf.reshape(tf.range(100), [10, 10])
+    >>> tensor_two = tf.range(10)
+    >>> formatted = tf.strings.format("first: {}, second: {}, suffix",
+    ...                               (tensor_one, tensor_two))
+    >>> print(formatted)
```
```suggestion
    >>> tf.strings.format("first: {}, second: {}, suffix",
    ...                               (tensor_one, tensor_two))
```

mihaimaruseac committed:
Update tensorflow/python/ops/string_ops.py

mihaimaruseac committed:
Update tensorflow/python/ops/string_ops.py

mihaimaruseac reviewed (dismissed):
Thank you

yongtang committed:
fix doctest failure

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

gbaned commented:
@yongtang Can you please address Ubuntu Sanity errors? Thanks!

yongtang committed:
Remove unneeded "```" in docstring as doctest does not use "```" to locate code block (use ">>>" instead)

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @mihaimaruseac @gbaned for the help. I have updated the PR. The previous failure was caused by the fact that doctest does not use "```" to locate the code block anymore (use ">>>" and "...") instead. I have removed the redundant "```" that was causing the test failure in the PR.

Please take a look and let me know if there are any other issues.

mihaimaruseac commented on tensorflow/python/ops/string_ops.py:
````diff
@@ -122,37 +123,19 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
-        assert(out.decode() == expected)
-    ```
+
+    >>> tensor_one = tf.reshape(tf.range(100), [10, 10])
````
Let's try a shorter range so that we don't need to have a line too long and disable pylint

yongtang committed:
Replace with a shorter example, and removes pylint disable

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @mihaimaruseac, the PR has been updated with a shorter example and the pylint disable has been removed as well.
